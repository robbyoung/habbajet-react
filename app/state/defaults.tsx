import {HabbajetEditor, PurchaseEditor} from '.';
import {habbajetColors} from '../colors';

const EMPTY_FIELD = {
    value: '',
    errorMessage: '',
};

export const DEFAULT_EDITOR_STATE: HabbajetEditor = {
    name: EMPTY_FIELD,
    value: EMPTY_FIELD,
    modifier: EMPTY_FIELD,
    slack: EMPTY_FIELD,
    color: habbajetColors[0],
    validated: false,
};

export const DEFAULT_PURCHASE_EDITOR_STATE: PurchaseEditor = {
    name: EMPTY_FIELD,
    cost: EMPTY_FIELD,
    tagId: '',
    validated: false,
};
