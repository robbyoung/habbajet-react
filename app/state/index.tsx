export enum HabitResult {
    Success,
    SlackSuccess,
    Failure,
}

export interface Habbajet {
    id: string;
    name: string;
    maxValue: number;
    currentValue: number;
    modifier: number;
    results: HabitResult[];
    color: string;
    date: string;
    toClaim: boolean;
    selected: boolean;
    currentStreak: number;
    bestStreak: number;
    oldStreaks: [number, number];
    totalSlack: number;
    remainingSlack: number;
}

export interface Purchase {
    id: string;
    name: string;
    cost: number;
    date: string;
    tagId: string;
}

export interface EditorField {
    value: string;
    errorMessage: string;
}

export interface HabbajetEditor {
    name: EditorField;
    value: EditorField;
    modifier: EditorField;
    slack: EditorField;
    color: string;
    validated: boolean;
}

export interface PurchaseEditor {
    name: EditorField;
    cost: EditorField;
    tagId: string;
    validated: boolean;
    id?: string;
}

export interface Tag {
    id: string;
    name: string;
    color: string;
}

export interface State {
    habbajets: Habbajet[];
    purchases: Purchase[];
    tags: Tag[];
    habbajetEditor: HabbajetEditor;
    purchaseEditor: PurchaseEditor;
    budget: number;
}
