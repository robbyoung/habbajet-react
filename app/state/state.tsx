export interface Habbajet {
    name: string;
    value: number;
    days: boolean[];
}

export interface Purchase {
    name: string;
    value: number;
}

export interface State {
    habits: Habbajet;
    purchages: Purchase;
    budget: string;
}
