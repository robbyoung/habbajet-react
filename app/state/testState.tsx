import {
    State,
    Habbajet,
    Purchase,
    HabbajetEditor,
    Tag,
    PurchaseEditor,
    TagEditor,
} from '.';
import {habbajetColors} from '../colors';

export function createTestState(
    numHabbajets: number,
    numPurchases: number,
    budget: number,
    selectedIndex: number = -1,
): State {
    const habbajets: Habbajet[] = [];
    for (let i = 0; i < numHabbajets; i++) {
        habbajets.push({
            id: `${i}`,
            name: `Habbajet ${i}`,
            maxValue: 100,
            currentValue: 100,
            modifier: 2,
            results: [],
            color: habbajetColors[i % habbajetColors.length],
            date: '2020-03-22T11:00:00.000Z',
            toClaim: false,
            selected: i === selectedIndex,
            currentStreak: 0,
            bestStreak: 5,
            oldStreaks: [0, 5],
            totalSlack: 0,
            remainingSlack: 0,
            description: '',
        });
    }

    const purchases: Purchase[] = [];
    for (let i = 0; i < numPurchases; i++) {
        purchases.push({
            id: `${i}`,
            name: `Purchase ${i}`,
            cost: i,
            date: '2020-03-22T11:00:00.000Z',
            tagId: `${i % 10}`,
        });
    }

    const tags: Tag[] = [];
    for (let i = 0; i < 10; i++) {
        tags.push({
            id: `${i}`,
            name: `Tag ${i}`,
            color: habbajetColors[i % habbajetColors.length],
        });
    }

    const EMPTY_FIELD = {
        value: '',
        errorMessage: '',
    };
    const habbajetEditor: HabbajetEditor = {
        name: {...EMPTY_FIELD},
        description: {...EMPTY_FIELD},
        value: {...EMPTY_FIELD},
        modifier: {...EMPTY_FIELD},
        slack: {...EMPTY_FIELD},
        color: habbajetColors[0],
        validated: false,
    };

    const purchaseEditor: PurchaseEditor = {
        name: {...EMPTY_FIELD},
        cost: {...EMPTY_FIELD},
        tagId: '',
        validated: false,
    };

    const tagEditor: TagEditor = {
        name: {...EMPTY_FIELD},
        color: habbajetColors[0],
        validated: false,
    };

    return {
        habbajets,
        purchases,
        budget,
        habbajetEditor,
        purchaseEditor,
        tagEditor,
        tags,
    };
}

export function createTestEditor(
    name: string,
    value: string,
    modifier: string,
    slack = '0',
    validated = false,
): HabbajetEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        description: {
            value: '',
            errorMessage: '',
        },
        value: {
            value,
            errorMessage: '',
        },
        modifier: {
            value: modifier,
            errorMessage: '',
        },
        slack: {
            value: slack,
            errorMessage: '',
        },
        color: habbajetColors[0],
        validated,
    };
}

export function createTestPurchaseEditor(
    name: string,
    value: string,
    validated = false,
): PurchaseEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        cost: {
            value,
            errorMessage: '',
        },
        tagId: '0',
        validated,
    };
}

export function createTestTagEditor(
    name: string,
    validated = false,
): TagEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        color: '#000000',
        validated,
    };
}
