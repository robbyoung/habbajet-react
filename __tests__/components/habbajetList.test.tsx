import React from 'react';
import renderer from 'react-test-renderer';
import {createTestState} from '../../app/state/testState';
import HabbajetList from '../../app/components/habbajetList';
import moment from 'moment';
import {render, fireEvent, wait} from '@testing-library/react-native';

describe('HabbajetList Component', () => {
    it('will create buttons out of an array of habbajets', () => {
        const habbajets = createTestState(30, 0, 0).habbajets;
        habbajets.forEach(habbajet => {
            habbajet.date = moment()
                .add(1, 'day')
                .toISOString();
        });
        const component = renderer.create(
            <HabbajetList habbajets={habbajets} onSelect={() => undefined} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will highlight habits with pending actions', () => {
        const habbajets = createTestState(30, 0, 0).habbajets;
        const component = renderer.create(
            <HabbajetList habbajets={habbajets} onSelect={() => undefined} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('can handle an empty habbajet array', () => {
        const habbajets = createTestState(0, 0, 0).habbajets;
        const component = renderer.create(
            <HabbajetList habbajets={habbajets} onSelect={() => undefined} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will run the onSelect callback if a habbajet button is pressed', async () => {
        const habbajets = createTestState(3, 0, 0).habbajets;
        const onSelect = jest.fn();
        const {getByTestId} = render(
            <HabbajetList habbajets={habbajets} onSelect={onSelect} />,
        );

        const habbajetButton = getByTestId('button-habbajet-1');
        fireEvent.press(habbajetButton);
        await wait(() => expect(onSelect).toBeCalledWith(habbajets[1]));
    });
});
