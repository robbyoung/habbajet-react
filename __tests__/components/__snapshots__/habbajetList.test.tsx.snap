// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabbajetList Component can handle an empty habbajet array 1`] = `
<View
  style={
    Object {
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#959595",
        "fontFamily": "Abel",
        "fontSize": 20,
        "margin": 0,
        "padding": 0,
      }
    }
  >
    My Habits
  </Text>
</View>
`;

exports[`HabbajetList Component will create buttons out of an array of habbajets 1`] = `
<View
  style={
    Object {
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#959595",
        "fontFamily": "Abel",
        "fontSize": 20,
        "margin": 0,
        "padding": 0,
      }
    }
  >
    My Habits
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#8066C2",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 0
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2AAD5E",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 1
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E04747",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 2
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7F90EC",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 3
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#B1B952",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 4
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 5
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#8066C2",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 6
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2AAD5E",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 7
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E04747",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 8
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7F90EC",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 9
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#B1B952",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 10
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 11
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#8066C2",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 12
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2AAD5E",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 13
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E04747",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 14
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7F90EC",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 15
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#B1B952",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 16
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 17
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#8066C2",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 18
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2AAD5E",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 19
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E04747",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 20
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7F90EC",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 21
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#B1B952",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 22
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 23
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#8066C2",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 24
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2AAD5E",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 25
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E04747",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 26
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7F90EC",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 27
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#B1B952",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 28
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 10,
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "Abel",
            "fontSize": 30,
            "padding": 10,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      Habbajet 29
    </Text>
  </View>
</View>
`;
